docs/source/guides/define_by_run.rst

```diff
@@ -0,0 +1,23 @@
+Define-by-run
+-------------
+
+.. currentmodule:: chainer
+
+As mentioned on the top page, Chainer is a flexible framework for neural networks.
+One major goal is flexibility, so it must enable us to write complex architectures simply and intuitively.
+
+Most existing deep learning frameworks are based on the **"Define-and-Run"** scheme.
+That is, first a network is defined and fixed, and then the user periodically feeds it with mini-batches.
+Since the network is statically defined before any forward/backward computation, all the logic must be embedded into the network architecture as *data*.
+Consequently, defining a network architecture in such systems (e.g. Caffe) follows a declarative approach.
+Note that one can still produce such a static network definition using imperative languages (e.g. torch.nn, Theano-based frameworks, and TensorFlow).
+
+In contrast, Chainer adopts a **"Define-by-Run"** scheme, i.e., the network is defined on-the-fly via the actual forward computation.
```

Comment:
How about _dynamically_ instead of _on-the-fly_, since the word _dynamically_ is used in below at line 20 it makes sense to relate these two!